internal/transactionsubmission/services/horizon.go

```diff
@@ -219,3 +226,95 @@ func DeleteChannelAccountOnChain(ctx context.Context, submiterEngine engine.Subm
 
 	return nil
 }
+
+// CreateAndFundAccount creates and funds a new destination account on the Stellar network with the given amount of native asset from the source account.
+func CreateAndFundAccount(ctx context.Context, submitterEngine engine.SubmitterEngine, amountNativeAssetToSend int, sourceAcc, destinationAcc string) error {
+	if sourceAcc == destinationAcc {
+		return fmt.Errorf("funding source account and destination account cannot be the same: %s", sourceAcc)
+	}
+
+	log.Ctx(ctx).Info("Initiating transaction...")
+	if err := retry.Do(func() error {
+		srcAccDetails, err := getAccountDetails(submitterEngine.HorizonClient, sourceAcc)
+		if err != nil {
+			return fmt.Errorf("getting details for source account: %w", err)
+		}
+
+		tx, err := txnbuild.NewTransaction(
+			txnbuild.TransactionParams{
+				SourceAccount:        srcAccDetails,
+				IncrementSequenceNum: true,
+				BaseFee:              int64(submitterEngine.MaxBaseFee),
+				Preconditions: txnbuild.Preconditions{
+					TimeBounds: txnbuild.NewTimeout(30),
+				},
+				Operations: []txnbuild.Operation{
+					&txnbuild.CreateAccount{
+						Destination: destinationAcc,
+						Amount:      strconv.Itoa(amountNativeAssetToSend),
+					},
+				},
+			},
+		)
+		if err != nil {
+			return fmt.Errorf(
+				"creating raw create account tx for account %s: %w",
+				destinationAcc,
+				err,
+			)
+		}
+		// Host distribution account signing:
+		tx, err = submitterEngine.HostAccountSigner.SignStellarTransaction(ctx, tx, sourceAcc)
+		if err != nil {
+			return fmt.Errorf(
+				"signing create account tx for account %s: %w",
+				destinationAcc,
+				err,
+			)
+		}
+
+		_, err = submitterEngine.HorizonClient.SubmitTransactionWithOptions(tx, horizonclient.SubmitTxOpts{SkipMemoRequiredCheck: true})
+		if err != nil {
+			return err
+		}
+
+		return nil
+	},
+		retry.Attempts(CreateAndFundAccountRetryAttempts),
+		retry.MaxDelay(1*time.Minute),
+		retry.DelayType(retry.BackOffDelay),
```

Comment:
@marwen-abid let me know what you think about this, and if they need to be tweaked!

Reply:
This seems great to me!